api/logbroker.proto

```diff
@@ -80,6 +80,9 @@ message LogMessage {
 
 	// Data is the raw log message, as generated by the application.
 	bytes data = 4;
+	// Details is a map of key value pairs representing additional log details
+	// that may have been returned from the task logger
+	map<string, string> details = 5; 
```

Comment:
nit: For consistency, there should be a blank line between `data` and this comment.

I'm wondering if we should make this a repeated field instead of a map. The map makes sense from a data model perspective. But I expect we'll be dealing with *lots* of log lines, and having a map for each one might use a lot of extra memory. A map is a hash table, so it will typically have lots of unused buckets that waste memory. I don't expect we'll be looking up particular details by key; rather we'll be iterating over all the keys to extract them from the map. Iteration is a lot cheaper with a repeated field. And if we're iterating over the keys to put them into a log line, if they come from a map we'd want to sort them, which would mean translating the map to a slice anyway.

Any thoughts?